ethereum/src/consensus.rs

```diff
@@ -357,17 +357,36 @@ impl<S: ConsensusSpec, R: ConsensusRpc<S>> Inner<S, R> {
 
         self.bootstrap(checkpoint).await?;
 
-        let current_period = calc_sync_period::<S>(self.store.finalized_header.beacon().slot);
-        let updates = self
+        let expected_current_slot = self.expected_current_slot();
```

Comment:
Also this update fetching has gotten a bit complex, so we should probably extract it to a `get_updates` function.